### PR TITLE
RTPS Discovery max_lease_duration config option

### PR DESCRIPTION
**Pull request by ClaytonCalabrese**
This PR creates a max_lease_duration option that limits what participants can set their lease duration to be.

I set the default at 2x the default lease duration (600 seconds).

### CODE REVIEW

File: dds/DCPS/RTPS/Spdp.cpp [hunk 1]
```diff
@@ -666,7 +666,16 @@ Spdp::handle_participant_data(DCPS::MessageId id,
         DCPS::LogGuid(guid_).c_str(), DCPS::LogGuid(guid).c_str(),
         pdata.leaseDuration.seconds, DCPS::LogAddr(from).c_str(), participants_.size()));
     }
-
+    ::CORBA::Long maxLeaseDuration = config_->max_lease_duration().to_dds_duration().sec;
+    if (pdata.leaseDuration.seconds > maxLeaseDuration) {
```
Comment (jrw972) [inline dds/DCPS/RTPS/Spdp.cpp:670]:
```suggestion
    if (maxLeaseDuration && pdata.leaseDuration.seconds > maxLeaseDuration) {
```

Comment (jrw972) [reply dds/DCPS/RTPS/Spdp.cpp:670]: This gives people a way to opt-out by setting it to 0.

**Review by jrw972 — CHANGES_REQUESTED** [review]

**Comment by ClaytonCalabrese** [comment]
**test_m10_o1d0_sec**

tests/DCPS/PersistentInfoRepo/run_test.pl
tests/DCPS/XTypes/run_test.pl
tests/security/attributes/run_test.pl SC0_sec_sub
tests/security/attributes/run_test.pl FullMsgEncrypt_SubMsgSign

**test_u20_p1_asan_sec**
tests/DCPS/EntityLifecycleStress/run_test.pl publishers 5 subscribers 5

Seem new. I'll look into the test failures as well as I can because I do not have access to a mac.

**Comment by ClaytonCalabrese** [comment]
Okay I think those tests might have just been a coincidence?

They haven't shown up again in the last 2 runs.

File: dds/DCPS/RTPS/Spdp.cpp [hunk 2]
```diff
@@ -666,7 +666,16 @@ Spdp::handle_participant_data(DCPS::MessageId id,
         DCPS::LogGuid(guid_).c_str(), DCPS::LogGuid(guid).c_str(),
         pdata.leaseDuration.seconds, DCPS::LogAddr(from).c_str(), participants_.size()));
     }
-
+    ::CORBA::Long maxLeaseDuration = config_->max_lease_duration().to_dds_duration().sec;
```
Comment (jrw972) [inline dds/DCPS/RTPS/Spdp.cpp:669]: This block should be conditional on !from_sedp (which would be a secure update).

**Review by jrw972 — CHANGES_REQUESTED** [review]

**Review by jrw972 — APPROVED** [review]

**Pull request merged**